app/pages/project/index.jsx

```diff
@@ -86,23 +88,17 @@ class ProjectPageController extends React.Component {
       }
     }
   }
+
   componentWillUnmount() {
     Split.clear();
   }
 
-  setupSplits(props) {
-    if (!props) {
-      ({ props } = this);
-    }
-    const { user } = props;
-    const { owner, name } = props.params;
-
+  setupSplits(slug, user) {
     if (user) {
-      Split.load(`${owner}/${name}`)
+      return Split.load(slug)
       .then((splits) => {
-        this.setState({ splits });
         if (!splits) {
-          return;
+          return null;
```

Comment:
If `splits` defaults to `{}` instead of `null`, that will simplify any logic that checks for `splits && splits[property]`. `ProjectHomeWorkflowButtons` defaults `splits` to `{}` so I think that change should be safe.

Reply:
Good idea.